src/main/java/com/comcast/zucchini/ZucchiniRuntime.java

```diff
@@ -197,14 +196,36 @@ public void run() throws IOException {
         TestContext tc = TestContext.getCurrent();
         AbstractZucchiniTest azt = tc.getParentTest();
 
+        Iterator<CucumberFeatureHolder> features = azt.fastrunIteratorFactory(this.ros.cucumberFeatures(this.rl).iterator());
+
         boolean parallel = azt.isParallel();
         boolean barrierEnabled = azt.canBarrier();
 
-        for(CucumberFeature cf : features) {
-            formatter.uri(cf.getPath());
-            formatter.feature(cf.getGherkinFeature());
+        CucumberFeatureHolder lastCFH = null;
+
+        while (true) {
+            CucumberFeatureHolder cfh = null;
+            try {
+                cfh = features.next();
+            } catch (NoSuchElementException ee) {
+                break;
+            }
+
+            CucumberFeature cf = cfh.getFeature();
+            CucumberTagStatement statement = cfh.getStatement();
 
-            for(CucumberTagStatement statement : cf.getFeatureElements()) {
+            if(    null == lastCFH
+                || lastCFH.getFeature() != cf) {
+                /* Setup a new feature IF the feature is different than the last time */
+                if (null != lastCFH) {
+                    /* Send the eof of a feature as we are on a new feature file */
+                    formatter.eof();
+                }
+                formatter.uri(cf.getPath());
+                formatter.feature(cf.getGherkinFeature());
+            }
+
+            if(null != statement) {
```

Comment:
This is needed.  We don't want to run the else condition if the statement is null.